crypto/param_build.c

```diff
@@ -233,8 +233,8 @@ static int push_BN(OSSL_PARAM_BLD *bld, const char *key,
 int OSSL_PARAM_BLD_push_BN(OSSL_PARAM_BLD *bld, const char *key,
                            const BIGNUM *bn)
 {
-    if (BN_is_negative(bn))
-        return push_BN(bld, key, bn, bn == NULL ? 0 : BN_num_bytes(bn) + 1,
+    if (bn != NULL && BN_is_negative(bn))
```

Comment:
This check is sane, certianly, but shouldn't we already be protected from this by the NULL check in OPENSSL_sk_push?

Reply:
`BN_is_negative()` is not safe for use with  NULL pointer

Reply:
@mattcaswell yeah, that makes sense, my question is: should we allow the placing of NULL pointers on a stack_of type variant in the first place?  All the uses of sk_<type>_num seem to assume that all items on a stack are non-null.  Would it be better to add a check to OPENSSL_sk_push to return a failure if the data pointer being pushed/inserted is NULL?

Reply:
This is a public API and it allows NULL BIGNUMs in 3.1/3.0 so this is necessary.